File: ui-api/src/main/java/io/tipsters/uiapi/web/CountriesController.java
package io.tipsters.uiapi.web;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

import io.tipsters.uiapi.dto.CountryData;
import io.tipsters.uiapi.service.CountryService;

@RestController
@RequestMapping("/countries")
public class CountriesController {

  private final CountryService countryService;

  public CountriesController(CountryService countryService) {
    this.countryService = countryService;
  }

  @GetMapping
  List<CountryData> countries(@RequestParam(name = "limit", defaultValue = "10") int limit) {
    return countryService.countriesOrderedByPriority(limit);
  }
}

File: web-server/src/main/kotlin/io/tipsters/controller/CountriesController.kt
package io.tipsters.controller

import io.tipsters.service.CountryService
import io.tipsters.service.data.CountryData
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod.GET
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/countries")
class CountriesController @Autowired constructor(private val countryService: CountryService) {

    @RequestMapping(method = arrayOf(GET))
    fun countries(@RequestParam(name = "limit", defaultValue = "10") limit: Int): List<CountryData> =
            countryService.countriesOrderedByPriority(limit)
}
